src/components/hmi_message_handler/src/websocket_session.cc

```diff
@@ -0,0 +1,332 @@
+         } else if (isResponse(message)) {
+            std::string id = message["id"].asString();
+            std::string method = findMethodById(id);
+            DBG_MSG(("Message is response on: %s\n", method.c_str()));
+            if ("" != method) {
+               if ("MB.registerComponent" == method) { 
+                  if (message.isMember("result") && message["result"].isInt())
+                  {
+                     mControllersIdStart = message["result"].asInt();
+                  } else
+                  {
+                     DBG_MSG_ERROR(("Not possible to initialize mControllersIdStart!\n"));
+                  }
+               } else if ("MB.subscribeTo" == method || "MB.unregisterComponent" == method || "MB.unsubscribeFrom" == method)
+               {
+                  //nothing to do for now
+               } else
+               {
+                  controller_->processResponse(method, message);
+               }
+            } else {
+               DBG_MSG_ERROR(("Request with id %s has not been found!\n", id.c_str()));
+            }
+         } else {
+            DBG_MSG(("Message is request!\n"));
+            std::string method = message["method"].asString();
+            std::string component_name = GetComponentName(method);
+
+            if(component_name == "MB") {
+               controller_->processInternalRequest(message, this);
+            } else {
+               controller_->pushRequest(message, this);
+               controller_->processRequest(message);
+            }
+         }
+      } else {
+         DBG_MSG_ERROR(("Message contains wrong data!\n"));
+         sendJsonMessage(error);
+      }         
+   }
+
+   bool WebsocketSession::isNotification(Json::Value& root) {
+      DBG_MSG(("CMessageBrokerController::isNotification()\n"));
+      bool ret = false;
+      if (false == root.isMember("id"))
+      {
+         ret = true;
+      }
+      DBG_MSG(("Result: %d\n", ret));
+      return ret;
+   }
+
+   bool WebsocketSession::isResponse(Json::Value& root) {
+      DBG_MSG(("CMessageBrokerController::isResponse()\n"));
+      bool ret = false;
+      if ((true == root.isMember("result")) || (true == root.isMember("error")))
+      {
+         ret = true;
+      }
+      DBG_MSG(("Result: %d\n", ret));
+      return ret;
+   }
+
+   std::string WebsocketSession::findMethodById(std::string id) {
+      DBG_MSG(("CMessageBrokerController::findMethodById()\n"));
+      sync_primitives::AutoLock auto_lock(queue_lock_);
+      std::string res = "";
+      std::map <std::string, std::string>::iterator it;
+      it = mWaitResponseQueue.find(id);
+      if (it != mWaitResponseQueue.end())
+      {
+         res = (*it).second;
+         mWaitResponseQueue.erase(it);
+      }
+      return res;
+   }
+
+   bool WebsocketSession::checkMessage(Json::Value& root, Json::Value& error)
+   {
+      DBG_MSG(("CMessageBrokerController::checkMessage()\n"));
+      Json::Value err;
+
+      try
```

Comment:
We've got to remove these try/catch statements. This project does not use them because of the [Google C++ Style Guide](https://google.github.io/styleguide/cppguide.html#Exceptions)